GpxExport/Models/Workout+FitWriter.swift

```diff
@@ -42,74 +42,80 @@ extension Workout {
                 return
             }
 
-            if let header = self.gpxHeader(title: self.name, startDate: self.startDate).data(using: .utf8) {
-                file.write(header)
-            }
+            let time = FitTime(date: Date())
+            let duration = Measurement(value: self.duration, unit: UnitDuration.seconds)
+            var messages: [FitMessage] = []
+            let activity = ActivityMessage(timeStamp: time,
+                                           totalTimerTime: duration,
+                                           localTimeStamp: nil,
+                                           numberOfSessions: nil,
+                                           activity: Activity.manual,
+                                           event: nil,
+                                           eventType: nil,
+                                           eventGroup: nil)
+            messages.append(activity)
+            let fileId = FileIdMessage(deviceSerialNumber: nil,
+                                        fileCreationDate: time,
+                                        manufacturer: .garmin,
+                                        product: nil,
+                                        fileNumber: nil,
+                                        fileType: FileType.activity,
+                                        productName: nil)
+
```

Comment:
Couple of things I would recommend.
Always have a Serial Number even it if is 1 same thing for File Number.